### ISSUE TITLE
Clearing metrics data without removing the metric

### ISSUE THREAD

**Issue by mwho** [siimon/prom-client#145]
Every time my metric is updated, I'd like to clear all existing data before doing the update. This is because the existing data might be stale, and I want to make sure that the data associated with the metric is fresh.

It seems like the only way to do this is something like the following - removing the metric and then creating it again.

```
function updateMyGauge(x,y,z,val) {
    register.removeSingleMetric('myMetric');
    myGauge = new promClient.Gauge({name: 'myMetric',
                                           help: 'myMetric_help',
                                           labelNames: ['label1',
                                                        'label2',
                                                        'label3']});
     myGauge.set({label1: x, label2: y, label3: z}, val)
}
```

Is there a better way? Otherwise, it'd be nice to have a method `register.clearMetricData` that clears all labels to value mappings but keeps the metric definition itself intact.

**Comment by siimon** [comment]
I've not used timestamps in prometheus myself, but that might be a way forward

**Comment by SimenB** [comment]
Using timestamps should work. I think it then has a 5 minute window before being discarded. Is that enough?

**Comment by mwho** [comment]
The sense I'm getting from reading the Prometheus forums (i.e. https://github.com/prometheus/client_golang/issues/187, https://github.com/prometheus/collectd_exporter/issues/24) is that specifying timestamps for metrics is discouraged. The Prometheus philosophy seems to be that the client side endpoint should always provide truthful and up to date data at time of scrape.

**Comment by SimenB** [comment]
This was fixed in #157
